package com.funcxy.oj.services;

/**
 * Created by aak12 on 2017/3/1.
 */

//@RunWith(SpringJUnit4ClassRunner.class)  //使用junit4进行测试
//@SpringBootTest(classes = Application.class)
public class UserServiceTest {
//    static User user = new User();
//    static String usernameValid = "zccz14";
//    static String usernameDuplicated = "  z ccz 1  4 ";
//    static String usernameEmpty = "      ";
//    static String emailValid = "hell@funcxy.com";
//    static String emailEmpty = "";
//    static String emailInvalid = "asfdjklas@.com";
//    static String passwordValid = "abc6789067890";
//    static String passwordEmpty = "";
//    static String passwordInvalid = "243";
//
//    @Before
//    public void validUser() throws Exception{
//        user.setUserId(ObjectId.get());
//        user.setUsername(usernameValid);
//        user.setEmail(emailValid);
//        user.setPassword(passwordValid);
//    }
//
//    // All valid
//    @Test
//    public void signUpTest0() throws Exception{
//        userService.save(user);
//    }
//
//    // Duplicated username
//    @Test(expected = Exception.class)
//    public void signUpTest1() throws Exception{
//        userService.save(user);
//        userService.save(user);
//    }
//    // Duplicated username
//    @Test(expected = Exception.class)
//    public void signUpTest2() throws Exception{
//        userService.save(user);
//        user.setUsername(usernameDuplicated);
//        userService.save(user);
//    }
//    // Empty username
//    @Test(expected = Exception.class)
//    public void signUpTest3() throws Exception{
//        user.setUsername(usernameEmpty);
//        userService.save(user);
//    }
//
//    // Empty email
//    @Test(expected = Exception.class)
//    public void signUpTest4() throws Exception{
//        user.setEmail(emailEmpty);
//        userService.save(user);
//    }
//    // Invalid email
//    @Test(expected = Exception.class)
//    public void signUpTest5() throws Exception{
//        user.setEmail(emailInvalid);
//        userService.save(user);
//    }
//
//    // Empty password
//    @Test(expected = Exception.class)
//    public void signUpTest6() throws Exception{
//        user.setPassword(passwordEmpty);
//        userService.save(user);
//    }
//    // Invalid password
//    @Test(expected = Exception.class)
//    public void signUpTest7() throws Exception{
//        user.setPassword(passwordInvalid);
//        userService.save(user);
//    }

}
